### ISSUE TITLE
App crash on iOS when navigating by heading from title to CollectionView with VoiceOver [A11Y]

### ISSUE THREAD

**Issue by towebo** [dotnet/maui#31371]
### Description

When focus is on the title and you try to navigate to the next heading it first fails and VoiceOver says that there's no next heading. If you then swipe right to go to the next control it crashes.
If you're somewhere far down in the list and navigate backwards by heading you'll end up with the VoiceOver cursor focusing the title because it has SemanticProperties.HeadingLevel set to some level. Because of this it's easy to trip into this trap.
You can avoid it by first tapping the collection view and then navigate by heading but this is an extra step that's quite easy to miss.
If it's hard to create a total fix it would be great if it at least doesn't crash.

This is not the case with TalkBack on Android.

### Steps to Reproduce

1. Clone the provided sample app, adjust the app id so it matches your provisioning profile.
2. Connect an iPhone to your machine so you can debug the app on it.
3. Activate VoiceOver.
4. Run the application on your iPhone.
5. Scroll down the list with items so the first heading "Heading 1" has gone off the screen.
6. Tap on the title "Home" so the VoiceOver cursor focuses it and reads it.
7. Ensure the rotor is set to Headings and swipe down to go to the next heading, that should be in the list.
8. VoiceOver says "Heading not found" and the VoiceOver cursor is still on the title.
9. Swipe from left to right to go to the next control which is the list.
10. A null exception is thrown and the app crashes.

Expected behavior
VoiceOver should be able to find the next heading present either visible on the screen among the items or scroll the collection view until it finds the next item marked as having a heading level Level1 or greater.

Actual behavior
A null exception is thrown from deep inside and the app crashes.

### Link to public reproduction project repository

https://github.com/towebo/HeadingNavigationCrash.git

### Version with bug

9.0.100 SR10

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Affected platforms

iOS

### Affected platform versions

iOS 18.6

### Did you find any workaround?

I haven't looked for any coding workarounds because I don't think it's possible. You can catch the exception in the program.cs in platform for iOS but what good does this? It's on its way out anyway.
If you perform the steps including 8 above and instead of swiping right to go to the next control you put the VoiceOver cursor in the list by tapping the on the list things is initiated correctly because from there you can navigate by heading both forwards and backwards.
Objects are probably initiated correctly when doing this.

### Relevant log output

```shell
Object reference not set to an instance of an object.

Stack Trace:
at Microsoft.Maui.Controls.Handlers.Items.GroupableItemsViewController`1[[Microsoft.Maui.Controls.ReorderableItemsView, Microsoft.Maui.Controls, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].UpdateDefaultSupplementaryView(DefaultCell cell, NSString elementKind, NSIndexPath indexPath)
   at Microsoft.Maui.Controls.Handlers.Items.GroupableItemsViewController`1[[Microsoft.Maui.Controls.ReorderableItemsView, Microsoft.Maui.Controls, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].GetViewForSupplementaryElement(UICollectionView collectionView, NSString elementKind, NSIndexPath indexPath)
   at Microsoft.Maui.Controls.Handlers.Items.GroupableItemsViewController`1[[Microsoft.Maui.Controls.ReorderableItemsView, Microsoft.Maui.Controls, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].__IRegistrarGenericTypeProxy__Microsoft_Maui_Controls_Handlers_Items_GroupableItemsViewController_1___GetViewForSupplementaryElement(IntPtr sel, IntPtr p0, IntPtr p1, IntPtr p2, IntPtr* exception_gchandle)
--- End of stack trace from previous location ---
   at ObjCRuntime.Runtime.ThrowException(IntPtr gchandle) in /Users/builder/azdo/_work/2/s/macios/src/ObjCRuntime/Runtime.cs:line 2609
   at UIKit.UIApplication.UIApplicationMain(Int32 argc, String[] argv, IntPtr principalClassName, IntPtr delegateClassName) in /Users/builder/azdo/_work/2/s/macios/src/UIKit/UIApplication.cs:line 72
   at UIKit.UIApplication.Main(String[] args, Type principalClass, Type delegateClass) in /Users/builder/azdo/_work/2/s/macios/src/UIKit/UIApplication.cs:line 104
   at HeadingNavigationCrash.Program.Main(String[] args) in C:\\src\\HeadingNavigationCrash\\HeadingNavigationCrash\\Platforms\\iOS\\Program.cs:line 15
```

**Comment by jfversluis** [comment]
Can you please try the new handler for CollectionView and see if the issue is then still there? https://learn.microsoft.com/dotnet/maui/whats-new/dotnet-9?view=net-maui-9.0#collectionview-and-carouselview

Additionally you mention connecting a physical device. Does this also happen with the Simulator?

**Comment by towebo** [comment]
You can't use VoiceOver on a simulator as far as I know so that's a non issue.

The new handlers resolved the issue. Thumbs up!
